src/puppetlabs/trapperkeeper/bootstrap.clj

```diff
@@ -132,26 +136,102 @@
             (str "Unable to find bootstrap.cfg file via --bootstrap-config "
                  "command line argument, current working directory, or on classpath")))))
 
-(schema/defn chain-files :- [schema/Str]
-  "Takes a list of files, reads all their lines in, and returns a flattened seq
-   of all their lines put together.
+(schema/defn indexed
+  "Returns seq of [index, item] pairs
+  [:a :b :c] -> ([0 :a] [1 :b] [2 :c])"
+  [coll]
+  (map vector (iterate inc 0) coll))
 
-   In the case of an empty file, nil would be returned by line-seq, so we remove
-   all the nils at the end"
-  [files :- [(schema/protocol io/IOFactory)]]
-  (remove nil? (flatten (map #(line-seq (io/reader %)) files))))
+(schema/defn get-annotated-bootstrap-entries :- [AnnotatedBootstrapEntry]
+  [configs :- [(schema/protocol io/IOFactory)]]
+  (flatten (for [config configs]
+             (for [[line-number line-text] (indexed (map remove-comments (line-seq (io/reader config))))
+                   :when (not (empty? line-text))]
+               {:bootstrap-file (str config)
```

Comment:
If we are actually supporting something coming from inside of the jar file, I think what we'd want to use here is the resource URI, which would end up looking something like `classpath:org/my/package/resource.extension`.

I haven't stepped through the full stack of the code, but definitely at the time when you parse the CLI args, you know they are Strings, and if it's possible for those raw Strings to be passed directly into this fn, then I think it's safe to use `schema/Str` for the schema.

In the future we might end up needing to pre-process some of these strings to make them into an `IOFactory` before we call `io/reader` on them - e.g. if we need to load a resource out of a jar - but I think that would happen in a new function called from inside of here, and that fn might take an `IOFactory` as input and call `io/reader` on it or something?